GovernmentPortal/Officials/OfficialExplorerControl.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GovernmentPortal.Models;
+using MesaSuite.Common.Data;
+using MesaSuite.Common.Extensions;
+using MesaSuite.Common.Utility;
+
+namespace GovernmentPortal.Officials
+{
+    public partial class OfficialExplorerControl : UserControl, IExplorerControl<Official>
+    {
+        public event EventHandler IsDirtyChanged;
+        public long GovernmentID { get; set; }
+
+        private bool _suppressDirty;
+        public OfficialExplorerControl()
+        {
+            InitializeComponent();
+        }
+
+        private bool _isDirty;
+        public bool IsDirty
+        {
+            get => _isDirty;
+            set
+            {
+                _isDirty = value;
+                IsDirtyChanged?.Invoke(this, new EventArgs());
+            }
+        }
+
+        public Official Model { get; set; }
+
+        private frmGenericExplorer<Official> _explorer = null;
+        public frmGenericExplorer<Official> Explorer { set => _explorer = value; }
+
+        public async void OnDeleteClicked()
+        {
+            if (Model == null)
+            {
+                MessageBox.Show("Cannot delete an unsaved Official.", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return;
+            }
+
+            if (MessageBox.Show("Are you sure you want to delete this Official?", "Warning", MessageBoxButtons.YesNo, MessageBoxIcon.Warning) != DialogResult.Yes)

```

Comment:
Minor suggestion:
Change this text to "Are you sure you want to impeach this Official?" or "Are you sure you want to fire this Official?"